tests/test_otel.py

```diff
@@ -257,8 +257,37 @@ def test_batches(client, trace_service, batch_count):
     assert len(trace_service.batches) == batch_count
 
     for batch in trace_service.batches:
-        assert get_attr(batch[0].resource, "service.name") == "test_service"
+        assert (
+            get_attr(batch[0].resource, "service.name")
+            == "unknown_service:nginx"
+        )
         assert len(batch[0].scope_spans[0].spans) == batch_size
 
     time.sleep(0.3)  # wait for +1 request to be flushed
     trace_service.batches.clear()
+
+
+@pytest.mark.parametrize(
+    "nginx_config",
+    [
+        {
+            "res_attrs": """
+                otel_service_name "test_service";
+                otel_resource_attr my.name "my name";
+                otel_resource_attr my.service "my service";
+            """,
+        }
+    ],
+    indirect=True,
+)
+def test_custom_resource_attributes(client, trace_service):
+    assert client.get("http://127.0.0.1:18080/ok").status_code == 200
+
+    trace_service.wait_one_batch()
```

Comment:
I still think, we need a get_batch(), so that tests individual tests don't have to worry about clearing or iterating. `test_batches` will remain an exception for now.

Reply:
OK, implemented get_batch.